node-js-basics-101/node_modules/engine.io-client/engine.io.js

```diff
@@ -0,0 +1,4660 @@
+	    passphrase: options.passphrase || this.passphrase,
+	    cert: options.cert || this.cert,
+	    ca: options.ca || this.ca,
+	    ciphers: options.ciphers || this.ciphers,
+	    rejectUnauthorized: options.rejectUnauthorized || this.rejectUnauthorized,
+	    perMessageDeflate: options.perMessageDeflate || this.perMessageDeflate,
+	    extraHeaders: options.extraHeaders || this.extraHeaders,
+	    forceNode: options.forceNode || this.forceNode,
+	    localAddress: options.localAddress || this.localAddress,
+	    requestTimeout: options.requestTimeout || this.requestTimeout,
+	    protocols: options.protocols || void 0
+	  });
+
+	  return transport;
+	};
+
+	function clone(obj) {
+	  var o = {};
+	  for (var i in obj) {
+	    if (obj.hasOwnProperty(i)) {
+	      o[i] = obj[i];
+	    }
+	  }
+	  return o;
+	}
+
+	/**
+	 * Initializes transport to use and starts probe.
+	 *
+	 * @api private
+	 */
+	Socket.prototype.open = function () {
+	  var transport;
+	  if (this.rememberUpgrade && Socket.priorWebsocketSuccess && this.transports.indexOf('websocket') !== -1) {
+	    transport = 'websocket';
+	  } else if (0 === this.transports.length) {
+	    // Emit error on next tick so it can be listened to
+	    var self = this;
+	    setTimeout(function () {
+	      self.emit('error', 'No transports available');
+	    }, 0);
+	    return;
+	  } else {
+	    transport = this.transports[0];
+	  }
+	  this.readyState = 'opening';
+
+	  // Retry with the next transport if the transport is disabled (jsonp: false)
+	  try {
+	    transport = this.createTransport(transport);
+	  } catch (e) {
+	    this.transports.shift();
+	    this.open();
+	    return;
+	  }
+
+	  transport.open();
+	  this.setTransport(transport);
+	};
+
+	/**
+	 * Sets the current transport. Disables the existing one (if any).
+	 *
+	 * @api private
+	 */
+
+	Socket.prototype.setTransport = function (transport) {
+	  debug('setting transport %s', transport.name);
+	  var self = this;
+
+	  if (this.transport) {
+	    debug('clearing existing transport %s', this.transport.name);
+	    this.transport.removeAllListeners();
+	  }
+
+	  // set up transport
+	  this.transport = transport;
+
+	  // set up transport listeners
+	  transport.on('drain', function () {
+	    self.onDrain();
+	  }).on('packet', function (packet) {
+	    self.onPacket(packet);
+	  }).on('error', function (e) {
+	    self.onError(e);
+	  }).on('close', function () {
+	    self.onClose('transport close');
+	  });
+	};
+
+	/**
+	 * Probes a transport.
+	 *
+	 * @param {String} transport name
+	 * @api private
+	 */
+
+	Socket.prototype.probe = function (name) {
+	  debug('probing transport "%s"', name);
+	  var transport = this.createTransport(name, { probe: 1 });
+	  var failed = false;
+	  var self = this;
+
+	  Socket.priorWebsocketSuccess = false;
+
+	  function onTransportOpen() {
+	    if (self.onlyBinaryUpgrades) {
+	      var upgradeLosesBinary = !this.supportsBinary && self.transport.supportsBinary;
```

Comment:
__[jshint]__
* If a strict mode function is executed using function invocation, its 'this' value will be undefined.

<!-- Comment by [SideCI](https://sideci.com) -->